src/ASM/BasisFunctionVals.h

```diff
@@ -29,4 +29,28 @@ struct BasisFunctionVals
   Matrix4D d3Ndu3; //!< Third order derivatives of basis functions
 };
 
+
+//! \brief Convenience type alias
+using BasisValuesPtrs = std::vector<const BasisFunctionVals*>;
+
+
+//! \brief Utility class holding a vector of basis function values.
```

Comment:
Here I had to pause a minute to understand what this class was about. Seems like the idea is to have a class which is a container holding BasisFunctionVals objects, which also can be cast into a an array of _pointers_ to the same BasisFunctionVals objects. Maybe describe this fact and why it is needed in a `\details` doxy section.

Reply:
i can do that. this is because we get pointers (references) back from the cache, but we don't want deal with the pointers in other usages. but i still need to pass pointers into fe-methods. this utility class handles that transparently through the casting operator.